apps/client/src/features/app-settings/panel/project-panel/ProjectListItem.tsx

```diff
@@ -0,0 +1,165 @@
+}
+
+export default function ProjectListItem({
+  current,
+  createdAt,
+  editingFilename,
+  editingMode,
+  filename,
+  onSubmit,
+  onToggleEditMode,
+  updatedAt,
+}: ProjectListItemProps) {
+  const [submitError, setSubmitError] = useState<string | null>(null);
+
+  const handleRefetch = async () => {
+    await ontimeQueryClient.invalidateQueries({ queryKey: PROJECT_LIST });
+  };
+
+  const handleSubmitRename = async (values: RenameProjectFormValues) => {
+    try {
+      if (!values.filename) {
+        setSubmitError('Filename cannot be blank');
+        return;
+      }
+      await renameProject(filename, values.filename);
+      await handleRefetch();
+      onSubmit?.();
+    } catch (error) {
+      if (error instanceof AxiosError) {
+        const errorMessage = error?.response?.data?.message;
+
+        setSubmitError(errorMessage);
+      } else {
+        setSubmitError('An unknown error occurred');
+      }
+    }
+  };
+
+  const handleSubmitDuplicate = async (values: DuplicateProjectFormValues) => {
+    try {
+      if (!values.newFilename) {
+        setSubmitError('Filename cannot be blank');
+        return;
+      }
+      await duplicateProject(filename, values.newFilename);
+      await handleRefetch();
+      onSubmit?.();
+    } catch (error) {
+      if (error instanceof AxiosError) {
+        const errorMessage = error?.response?.data?.message;
+
+        setSubmitError(errorMessage);
+      } else {
+        setSubmitError('An unknown error occurred');
+      }
+    }
+  };
+
+  const handleCancel = () => {
+    onToggleEditMode?.(null, null);
+  };
+
+  useEffect(() => {
+    setSubmitError(null);
+  }, [editingMode]);
+
+  const renderEditMode = useMemo(() => {
+    switch (editingMode) {
+      case 'rename':
+        return (
+          <RenameProjectForm
+            filename={filename}
+            onSubmit={handleSubmitRename}
+            onCancel={handleCancel}
+            submitError={submitError}
+          />
+        );
+      case 'duplicate':
+        return (
+          <DuplicateProjectForm
+            filename={filename}
+            onSubmit={handleSubmitDuplicate}
+            onCancel={handleCancel}
+            submitError={submitError}
+          />
+        );
+      default:
+        return null;
+    }
+  }, [editingMode, filename, submitError]);
+
+  return (
+    <tr key={filename} className={current ? style.current : undefined}>
+      <td>{editingMode && filename === editingFilename ? renderEditMode : <span>{filename}</span>}</td>
+      <td>{createdAt}</td>
+      <td>{updatedAt}</td>
+      <td className={style.actionButton}>
+        <ActionMenu filename={filename} onAction={handleRefetch} onChangeEditMode={onToggleEditMode} />
+      </td>
+    </tr>
+  );
+}
+
+function ActionMenu({
+  filename,
+  onAction,
+  onChangeEditMode,
+}: {
+  filename: string;
+  onAction?: () => void;
+  onChangeEditMode?: (editMode: EditMode, filename: string) => void;
```

Comment:
question: I am not sure why this functions are optional? there is no use case for this menu to exist without them